### PR TITLE
Create Oracle SQL Sessions Assignment # 1 (Chaitanya Waghaye)

### PR DESCRIPTION
**Pull request by Chaitanya1603**

### CODE REVIEW

**Review by Chaitanya1603 — COMMENTED** [review]
some words missing

**Review by bharatkumarraith — APPROVED** [review]

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 1]
```diff
@@ -0,0 +1,107 @@
+1. 	select COUNT(*) as total_employees from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:2]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+2. 	select COUNT(*) as Number_of_departmennts from dept;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:4]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+3. 	select * from emp;
+   	select count(*) as Ename from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:7]: ok. second query is not needed.

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+4.	select * from dept;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:9]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+5. 	select SUM(sal) as total_salary_paid from emp;
+
+6. 	select SUM(comm) as total_commision from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:13]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+7. 	select DISTINCT JOB from emp where COMM is NOT NULL;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:15]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+8. 	Select SYSDATE as current_system_date from DUAL;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:17]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+9.	select AVG(sal) avg_salary_paid_to_employee from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:19]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+10. 	select deptno, COUNT(empno) from emp
+	group by deptno
+	order by deptno;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:23]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+11.  	select deptno , sum(sal) from emp
+	group by deptno
+	order by deptno;
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:26]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+11.  	select deptno , sum(sal) from emp
+	group by deptno
+	order by deptno; 	
+
+12.	select job , count(empno) from emp
+	group by job
+	order by job;	
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:31]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+13.	select job , round(avg(sal),3) from emp
+	group by job
+	order by job;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:35]: ok. salary rounded to 3 decimal places do not have a meaning. 2 decimals is a correct way.

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+14.	select empno, ename,
+	to_char(hiredate, 'DD') as hire_day,
+	to_char (hiredate, 'MM') as hire_month,
+	to_char (hiredate, 'YYYY') as hire_year
+	from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:41]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+15.	select deptno, empno, ename
+	from emp
+	order by deptno, empno;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:45]: not ok. sort asked only on department wise

select * from emp order by deptno;

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+16.	select empno, ename , job
+	from emp
+	order by job , empno;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:49]: not ok. sort asked only on job titles.

select * from emp order by job;

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+	order by deptno, empno;
+
+16.	select empno, ename , job
+	from emp
+	order by job , empno;
+ 
+17.	select empno,ename,sal
+	from emp
+	order by sal desc;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:53]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+18.	select empno,ename,deptno,sal
+	from emp
+	order by deptno asc, sal desc, empno;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:57]: ok. ASC is default, can be ignored.

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+19.	select count(ename)
+	from emp
+	where Length(ename) = 6;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:61]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+20.	select max(sal) ,min(sal) 
+	from emp;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:64]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+21. 	select deptno , max(sal), min (sal) , round(avg(sal)) , sum(sal)
+	from emp
+	group by deptno
+	order by deptno;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:69]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+22.	select empno,ename,hiredate
+	from emp
+	order by hiredate;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:73]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+23.	select empno,ename,hiredate
+	from emp
+	order by hiredate desc
+	Fetch first 1 row only;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:78]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+24.	select empno,ename,hiredate
+	from emp
+	order by hiredate 
+	fetch first 1 row only;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:83]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+25.	select to_char(hiredate ,'YYYY')as hireyear,deptno
+	from emp
+	order by hiredate desc, deptno asc;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:87]: not ok. order by should have the expression. select columns do not bring employee data.

select * from emp order by to_char(hiredate,'yyyy') desc, deptno;

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 25]
```diff
@@ -0,0 +1,107 @@
...
+26.	select empno,ename,sal
+	from emp
+	where sal >= (select (avg(sal))from emp);
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:91]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+27.	select empno,ename,sal
+	from emp
+	where sal <= (select (avg(sal))from emp);
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:95]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 27]
```diff
@@ -0,0 +1,107 @@
...
+28.	select empno , ename , sal
+	from emp 
+	where sal between 2000 and 4000;
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:99]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 28]
```diff
@@ -0,0 +1,107 @@
...
+29.	select empno , ename , sal
+	from emp
+	where sal= (select max(sal) from emp)
+	or sal = (select min (sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:104]: ok

File: Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1 [hunk 29]
```diff
@@ -0,0 +1,107 @@
...
+30.     select count (*) as no_of_gifts
+	from emp
+	where to_char(hiredate ,'MM') = to_char(Add_MONTHS(sysdate,1),'MM');
```
Comment (raviproengage) [inline Assignments/Chaitanya/Oracle SQL Sessions Assignment # 1:107]: ok

**Review by raviproengage — COMMENTED** [review]
Good Submission. Keep it up!

**Pull request merged**